matchzoo/data_pack/data_pack.py

```diff
@@ -294,6 +294,32 @@ def drop_label(self):
         """
         self._relation = self._relation.drop(columns='label')
 
+    @_optional_inplace
+    def drop_invalid(self):
+        """
+        Remove rows from the data pack where the length is zero.
+
+        :param inplace: `True` to modify inplace, `False` to return a modified
+            copy. (default: `False`)
+
+        Example:
+            >>> import matchzoo as mz
+            >>> data_pack = mz.datasets.toy.load_data()
+            >>> data_pack.append_text_length(inplace=True, verbose=0)
+            >>> data_pack.drop_invalid(inplace=True)
+        """
+        if not ('length_left' in self._left and 'length_right' in self._right):
+            raise ValueError(f"`length_left` or `length_right` is missed. "
+                             f"Please call `append_text_length` in advance.")
+        valid_left = self._left.loc[self._left.length_left != 0]
+        valid_right = self._right.loc[self._right.length_right != 0]
+        self._left = self._left[self._left.index.isin(valid_left.index)]
+        self._right = self._right[self._right.index.isin(valid_right.index)]
+        self._relation = self._relation[self._relation.id_left.isin(
+            valid_left.index) & self._relation.id_right.isin(
+                valid_right.index)]
+        self._relation.reset_index(drop=True, inplace=True)
+
```

Comment:
the code is difficult to read, can we follow "join" -> 'process" -> "split" pipeline?

1. Join three dataframes into big one as pandas dataframe.
2. Filter columns based on conditions, such as `f = f.loc[(f['length_left'] > 0) & (f['length_right'] > 0)]` and reset index.
3. Split big dataframe into `left`, `right`, `relation` again.

I guess it's more readable and safe.

Reply:
I think the code now is acceptable.